utils/ccdatofhir/mapping_resource_immunization.bal

```diff
@@ -0,0 +1,262 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// --------------------------------------------------------------------------------------------#
+// Source C-CDA to FHIR - Resource Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerina/log;
+import ballerinax/health.fhir.r4;
+
+# Map CCDA Immunization Activity to FHIR Immunization.
+#
+# + substanceAdministrationElement - CCDA Immunization Activity Element
+# + return - FHIR Immunization Resource
+public isolated function mapCcdaImmunizationToFhir(xml substanceAdministrationElement) returns r4:Immunization? {
+    if isXMLElementNotNull(substanceAdministrationElement) {
+        r4:Immunization immunization = {patient: {}, occurrenceDateTime: "", occurrenceString: "", vaccineCode: {}, status: "not-done"};
+
+        xml idElement = substanceAdministrationElement/<v3:id|id>;
+        xml statusCodeElement = substanceAdministrationElement/<v3:statusCode|statusCode>;
+        xml effectiveTimeElement = substanceAdministrationElement/<v3:effectiveTime|effectiveTime>;
+        xml repeatNumberElement = substanceAdministrationElement/<v3:repeatNumber|repeatNumber>;
+        xml routeCodeElement = substanceAdministrationElement/<v3:routeCode|routeCode>;
+        xml approachSiteCodeElement = substanceAdministrationElement/<v3:approachSiteCode|approachSiteCode>;
+        xml doseQuantityElement = substanceAdministrationElement/<v3:doseQuantity|doseQuantity>;
+        xml vaccineCodeElement = substanceAdministrationElement/<v3:consumable|consumable>/<v3:manufacturedProduct|manufacturedProduct>/<v3:manufacturedMaterial|manufacturedMaterial>/<v3:code|code>;
+        xml lotNumberTextElement = substanceAdministrationElement/<v3:consumable|consumable>/<v3:manufacturedProduct|manufacturedProduct>/<v3:manufacturedMaterial|manufacturedMaterial>/<v3:lotNumberText|lotNumberText>;
+        xml performerElement = substanceAdministrationElement/<v3:performer|performer>;
+        xml entryRelationship = substanceAdministrationElement/<v3:entryRelationship|entryRelationship>;
+
+        int index = 0;
+        foreach xml idElem in idElement {
+            r4:Identifier? mapCcdaIdToFhirIdentifierResult = mapCcdaIdToFhirIdentifier(idElem);
+            if mapCcdaIdToFhirIdentifierResult is r4:Identifier {
+                immunization.identifier[index] = mapCcdaIdToFhirIdentifierResult;
+                index = index + 1;
+            }
+        }
```

Comment:
Refer to : https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/pull/16/files#r1261079127

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/e973cad61936b398a3651cbc34f89d9f66c2aa15